clang/include/clang/Basic/Attr.td

```diff
@@ -1889,6 +1889,36 @@ def LifetimeBound : DeclOrTypeAttr {
   let SimpleHandler = 1;
 }
 
+def LifetimeCaptureBy : DeclOrTypeAttr {
+  let Spellings = [Clang<"lifetime_capture_by", 0>];
+  let Subjects = SubjectList<[ParmVar, ImplicitObjectParameter], ErrorDiag>;
+  let Args = [VariadicParamOrParamIdxArgument<"Params">];
+  let Documentation = [LifetimeCaptureByDocs];
+  let AdditionalMembers = [{
+private:
+  IdentifierInfo** ArgIdents;
+  SourceLocation* ArgLocs;
```

Comment:
Can we use `ArrayRef<IndentifierInfo*>` and `ArrayRef<SourceLocation>`? I'm not a big fan on using the raw pointers for arrays.

Reply:
Done.